File: surf-cloud-velocity/src/main/java/dev/slne/surf/cloud/velocity/SurfCloudVelocityInstance.java
package dev.slne.surf.cloud.velocity;

import com.google.auto.service.AutoService;
import dev.slne.surf.cloud.api.SurfCloudInstance;
import dev.slne.surf.cloud.core.SurfCloudCoreInstance;
import java.nio.file.Path;

@AutoService(SurfCloudInstance.class)
public final class SurfCloudVelocityInstance extends SurfCloudCoreInstance {

  public SurfCloudVelocityInstance() throws IllegalAccessException {
  }

  @Override
  public Path getDataFolder() {
    return null;
  }

  @Override
  public ClassLoader getClassLoader() {
    return null;
  }
}

File: surf-cloud-velocity/src/main/kotlin/dev/slne/surf/cloud/velocity/SurfCloudVelocityInstance.kt
package dev.slne.surf.cloud.velocity

import com.google.auto.service.AutoService
import dev.slne.surf.cloud.api.common.SurfCloudInstance
import dev.slne.surf.cloud.core.client.ClientCommonCloudInstance
import dev.slne.surf.cloud.core.common.coreCloudInstance
import dev.slne.surf.cloud.core.common.server.CommonCloudServerImpl
import dev.slne.surf.cloud.core.common.util.checkInstantiationByServiceLoader
import dev.slne.surf.cloud.velocity.listener.ListenerManager
import dev.slne.surf.cloud.velocity.netty.VelocityNettyManager

@AutoService(SurfCloudInstance::class)
class SurfCloudVelocityInstance : ClientCommonCloudInstance(VelocityNettyManager) {
    init {
        checkInstantiationByServiceLoader()
    }

    override suspend fun onEnable() {
        super.onEnable()

        ListenerManager.registerListeners()
    }

    override suspend fun onDisable() {
        super.onDisable()

        ListenerManager.unregisterListener()
    }
}

val velocityCloudInstance
    get() = coreCloudInstance as SurfCloudVelocityInstance
